src/java/main/org/apache/zookeeper/client/ZKClientConfig.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.client;
+
+import java.io.File;
+
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.common.ZKConfig;
+import org.apache.zookeeper.server.quorum.QuorumPeerConfig.ConfigException;
+
+/**
+ * Handles client specific properties
+ */
+public class ZKClientConfig extends ZKConfig {
+    public static final String ZK_SASL_CLIENT_USERNAME = "zookeeper.sasl.client.username";
+    public static final String ZK_SASL_CLIENT_USERNAME_DEFAULT = "zookeeper";
+    @SuppressWarnings("deprecation")
+    public static final String LOGIN_CONTEXT_NAME_KEY = ZooKeeperSaslClient.LOGIN_CONTEXT_NAME_KEY;;
+    public static final String LOGIN_CONTEXT_NAME_KEY_DEFAULT = "Client";
+    @SuppressWarnings("deprecation")
+    public static final String ENABLE_CLIENT_SASL_KEY = ZooKeeperSaslClient.ENABLE_CLIENT_SASL_KEY;
+    @SuppressWarnings("deprecation")
+    public static final String ENABLE_CLIENT_SASL_DEFAULT = ZooKeeperSaslClient.ENABLE_CLIENT_SASL_DEFAULT;
+    public static final String ZOOKEEPER_SERVER_REALM = "zookeeper.server.realm";
+    /**
+     * This controls whether automatic watch resetting is enabled. Clients
+     * automatically reset watches during session reconnect, this option allows
+     * the client to turn off this behavior by setting the property
+     * "zookeeper.disableAutoWatchReset" to "true"
+     */
+    public static final String DISABLE_AUTO_WATCH_RESET = "zookeeper.disableAutoWatchReset";
+    @SuppressWarnings("deprecation")
+    public static final String ZOOKEEPER_CLIENT_CNXN_SOCKET = ZooKeeper.ZOOKEEPER_CLIENT_CNXN_SOCKET;
+    /**
+     * Setting this to "true" will enable encrypted client-server communication.
+     */
+    @SuppressWarnings("deprecation")
+    public static final String SECURE_CLIENT = ZooKeeper.SECURE_CLIENT;
+
+    public ZKClientConfig() {
```

Comment:
Pro-tip: this constructor is not required, if you remove this constructor then calling
```
new ZKClientConfig()
```
will automatically call (the parent) ZKConfig's default constructor.